models/egraph_index_model.erl

```diff
@@ -242,34 +242,49 @@ create_or_update_info(Info, State) ->
     end.
 
 delete_resource(Key, KeyType, IndexName) ->
+    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
```

Comment:
You don't have to duplicate so much of code. Instead you could just write a new function to generate query and params instead and reuse most of it there.

```
delete_resource(Key, KeyType, IndexName) ->
    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
    delete_resource_internal(Query, Params).

delete_resource(Key, KeyType, Value, IndexName) ->
    {Query, Params} = form_delete_index_base_query(Key, KeyType, IndexName),
    Query1 = <<Query/binary, " and id=?">>,
    Params1 = Params ++ [Value],
    delete_resource_internal(Query1, Params1).

delete_resource_internal(Query, Params) ->
    TimeoutMsec = egraph_config_util:mysql_rw_timeout_msec(index),
    PoolName = egraph_config_util:mysql_rw_pool(index),
    case egraph_sql_util:mysql_write_query(
        PoolName, Query, Params, TimeoutMsec) of
        ok ->
            true;
        _ ->
            false
    end.
```